File: cef-web/src/main/java/gov/epa/cef/web/service/OperatingDetailService.java
package gov.epa.cef.web.service;

import gov.epa.cef.web.service.dto.OperatingDetailDto;

public interface OperatingDetailService {

    /**
     * Update an Operating Detail
     * @param dto
     * @return
     */
    OperatingDetailDto update(OperatingDetailDto dto);

}

File: cef-web/src/main/webapp/src/app/core/services/operating-detail.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { OperatingDetail } from 'src/app/shared/models/operating-detail';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class OperatingDetailService {

  private baseUrl = 'api/operatingDetail';  // URL to web api

  constructor(private http: HttpClient) { }

  update(detail: OperatingDetail): Observable<OperatingDetail> {
    const url = `${this.baseUrl}/${detail.id}`;
    return this.http.put<OperatingDetail>(url, detail);
  }

}
